java/src/hu/holyoil/skeleton/testcases/AsteroidExplodesWithTeleportGate.java

```diff
@@ -43,26 +48,15 @@ protected void load() {
         );
 
         settler.PlaceTeleporter();
+        settler.Move(asteroid2);
 
-        if (!isTeleportgateInStorage) {
-
-            Asteroid otherAsteroid = new Asteroid();
-            Logger.RegisterObject(otherAsteroid, "otherAsteroid: Asteroid");
-            t2.SetHomeAsteroid(otherAsteroid);
-
-            otherAsteroid.SetTeleporter(settler.GetStorage().GetOneTeleporter());
-            settler.GetStorage().RemoveTeleportGate(
-                    settler.GetStorage().GetOneTeleporter()
-            );
-
+        if (!isTeleportGateInStorage) {
+            settler.PlaceTeleporter();
```

Comment:
This is not good, because this asteroid already has a teleporter (see Line 50). Needs to be reverted.